webapp/src/marked.tsx

```diff
@@ -406,8 +407,17 @@ export class MarkedContent extends data.Component<MarkedContentProps, MarkedCont
                         // need to filter out editors that are currently hidden as we leave toolboxes in dom
                         const editorSelector = `#maineditor > div:not([style*="display:none"]):not([style*="display: none"])`;
                         const toolboxSelector = `${editorSelector} .blocklyTreeRow[data-ns="${ns}"]`;
-                        const toolboxRow = document.querySelector<HTMLDivElement>(toolboxSelector);
-                        toolboxRow?.click();
+                        let toolboxRow = document.querySelector<HTMLDivElement>(toolboxSelector);
+                        if (toolboxRow) {
+                            toolboxRow.click();
+                        } else if (isAdvanced) {
+                            // toggle advanced open first
+                            const advancedSelector = `${editorSelector} .blocklyTreeRow[data-ns="advancedcollapsed"]`;
+                            const advancedRow = document.querySelector<HTMLDivElement>(advancedSelector);
```

Comment:
Any reason we can't look up and expand the advanced row earlier (before the initial toolboxrow look up) since we already have isAdvanced, then reduce the duplication around looking up and clicking on the toolbox row?

Reply:
You're totally right, I initially had it as toggling advanced, fixed that, and forgot to refactor. Thanks for pointing out